Makefile

```diff
@@ -1,9 +1,35 @@
-# Go parameters
 GOCMD = go
 GOTEST = $(GOCMD) test 
 
+GOLANGCI_VERSION ?= v1.57.2
+TOOLS_BIN := $(shell mkdir -p build/tools && realpath build/tools)
+
+GOLANGCI = $(TOOLS_BIN)/golangci-lint-$(GOLANGCI_VERSION)
+$(GOLANGCI):
+	rm -f $(TOOLS_BIN)/golangci-lint*
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/$(GOLANGCI_VERSION)/install.sh | sh -s -- -b $(TOOLS_BIN) $(GOLANGCI_VERSION)
+	mv $(TOOLS_BIN)/golangci-lint $(TOOLS_BIN)/golangci-lint-$(GOLANGCI_VERSION)
+
 test:
 	$(GOTEST) ./...
 
-generate: $(esc)
-	$(GOCMD) generate
+validate: validate-lint validate-dirty ## Run validation checks.
```

Comment:
Is this necessary? Instead, we can use a build workflow to run the linter. Contributors should have their editor linter set up.

Reply:
Merging it as is, but happy to discuss further what this should look like across the org.